filesource.go

```diff
@@ -216,48 +216,53 @@ func (s *FileSource) runMergeFile() error {
 	}
 }
 
-func (s *FileSource) streamIncomingFile(newIncomingFile *incomingBlocksFile, blocksStore dstore.Store) error {
-	defer func() {
-		close(newIncomingFile.blocks)
-	}()
-
-	atomic.AddInt64(&currentOpenFiles, 1)
-	s.logger.Debug("open files", zap.Int64("count", atomic.LoadInt64(&currentOpenFiles)), zap.String("filename", newIncomingFile.filename))
-	defer atomic.AddInt64(&currentOpenFiles, -1)
-
-	// FIXME: Eventually, RETRY for this given file.. and continue to write to `newIncomingFile`.
-	reader, err := blocksStore.OpenObject(context.Background(), newIncomingFile.filename)
-	if err != nil {
-		return fmt.Errorf("fetching %s from block store: %w", newIncomingFile.filename, err)
-	}
-	defer reader.Close()
+type retryableError struct {
+	error
+}
 
-	blockReader, err := s.blockReaderFactory.New(reader)
-	if err != nil {
-		return fmt.Errorf("unable to create block reader: %w", err)
+func (e retryableError) Error() string {
+	return e.error.Error()
+}
+func retryable(err error) error {
+	return retryableError{
+		error: err,
 	}
+}
+func isRetryable(err error) bool {
+	_, ok := err.(retryableError)
+	return ok
+}
 
+func (s *FileSource) streamReader(blockReader BlockReader, prevLastBlockRead BlockRef, output chan *PreprocessedBlock) (lastBlockRead BlockRef, err error) {
+	prevPassed := prevLastBlockRead == nil
```

Comment:
I think it would be better to have `prevPassed` a bit bigger like `passedLastBlockSent` something along this. 

Also, to make it a bit clearer, I would always initialized `passedLastBlockSent` to `false` and when we do `if !prevPassed` I would do `if skipBlockBeforeRef == nil && !passedLastBlockSent`